modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala

```diff
@@ -546,6 +546,46 @@ object ElasticQuery {
   final def ids(value: String, values: String*): IdsQuery[Any] =
     Ids(values = Chunk.fromIterable(value +: values))
 
+  /**
+   * Constructs a type-safe instance of [[zio.elasticsearch.query.KNNQuery]] using the specified parameters.
+   * [[zio.elasticsearch.query.KNNQuery]] is used to perform a k-nearest neighbor (kNN) search and returns the matching
+   * documents.
+   *
+   * @param field
+   *   the type-safe field for which query is specified for
+   * @param k
+   *   number of nearest neighbors to return as top hits (must be less than `numCandidates`)
+   * @param numCandidates
+   *   number of nearest neighbor candidates to consider per shard
+   * @param queryVector
+   *   query vector
+   * @tparam S
+   *   document for which field query is executed
+   * @return
+   *   an instance of [[zio.elasticsearch.query.KNNQuery]] that represents the kNN query to be performed.
+   */
+  final def kNN[S](field: Field[S, _], k: Int, numCandidates: Int, queryVector: Chunk[Double]): KNNQuery[S] =
```

Comment:
Maybe we could rename `k` to something more meaningful.

Reply:
I guess `k` makes more sense to users using it. Additionally, it's a kNN search.